src/ocm/components/hosts/HostsDiscoveryTable.tsx

```diff
@@ -109,9 +89,23 @@ const HostsDiscoveryTable: React.FC<HostsDiscoveryTableProps> = ({ cluster }) =>
       disksColumn,
       countColumn(cluster),
     ],
-    [onEditHost, actionChecks.canEditHostname, actionChecks.canEditRole, onEditRole, cluster],
+    [onEditHost, actionChecks.canEditHostname, actionChecks.canEditRole, onEditRole, cluster, t],
   );
 
+  const expandMemo = React.useMemo(() => {
+    return function ExpandedHostDetail({ obj: host }: { obj: Host }) {
+      return (
+        <HostDetail
+          key={host.id}
+          host={host}
+          onDiskRole={onDiskRole}
+          canEditDisks={actionChecks.canEditDisks}
+          AdditionalNTPSourcesDialogToggleComponent={AdditionalNTPSourcesDialogToggle}
+        />
+      );
+    };
+  }, [onDiskRole, actionChecks.canEditDisks]);
```

Comment:
component inside component is an antipattern. It will break React's reconciliation loop.
Pass additional props to `HostDetail` could be done via context

Reply:
I've been looking at this but I've not been able to improve this.
From what I see, the `ExpandComponent` is forcing us to return this builder function which creates the HostDetail component.

I looked at if I could pass the actions from the context (from `useHostsTable`), but for that I'd need to pass the entire `cluster` object and then it would refresh when the cluster does.

I'm doing a slight change from the initial code, please let me know what you think.

Reply:
Just create your own context, wrap the table in it and load the content of the context in ExpandedComponent.

I know it is a bit too complicated but still better than passing all the actions and who knows what else would be needed for other cases

Reply:
It took me a while, but I think I managed to do as you suggested. PTAL.

Reply:
great, that is exactly what I wanted to see